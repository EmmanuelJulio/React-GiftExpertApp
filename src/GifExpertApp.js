import { type } from 'os'
import { listenerCount } from 'process'
import React,{ useState } from 'react'
import {AddCategory} from './components/AddCategory'
import { GrifGrid } from './components/GrifGrid'
//snipet para la creacion rapida de componente "rafc"
export const GifExpertApp = ( { defautlCategories= [] } ) => {
    // const categories=
    
    const [categories, setCategories] = useState( defautlCategories )
   
    // const handleAdd=()=>{
    //     // setCategories([...categories,'Hunter Hunter'])
    //     setCategories((cats=>[...categories,'Hunter hunter']))
    // }
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr/>
          
            <ol>
                {categories.map(category=> ( 
                <GrifGrid 
                key={category}
                category={category}></GrifGrid>
                ))
                }
                
            </ol>
        </>
    )
}
