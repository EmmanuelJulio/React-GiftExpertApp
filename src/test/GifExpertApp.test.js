import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { GifExpertApp } from "../GifExpertApp";


describe("Pruebas en GifexpertApp", () => {
  test("Debe de hacer match con el snapshot", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot(`""`);
  });

  test("Debe de mostrar una lista de categorias", () => {
    const categories = ['One punch', 'Dragon Ball'];
    const wrapper = shallow(<GifExpertApp defautlCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GrifGrid").length).toBe(categories.length);
  });
});
