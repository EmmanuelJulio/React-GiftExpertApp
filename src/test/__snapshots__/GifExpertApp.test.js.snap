// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifexpertApp Debe de hacer match con el snapshot: "" 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en GifexpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GrifGrid
      category="One punch"
      key="One punch"
    />
    <GrifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
